core/src/main/scala/org/http4s/otel4s/OtelMetrics.scala

```diff
@@ -0,0 +1,191 @@
+    */
+  def metricsOps[F[_]: Monad: Meter](
+      attributes: Attributes = Attributes.empty,
+      responseDurationSecondsHistogramBuckets: BucketBoundaries = DefaultHistogramBuckets,
+  ): F[MetricsOps[F]] =
+    for {
+      metrics <- createMetricsCollection(responseDurationSecondsHistogramBuckets)
+    } yield createMetricsOps(metrics, attributes)
+
+  private def createMetricsOps[F[_]](
+      metrics: MetricsCollection[F],
+      attributes: Attributes,
+  ): MetricsOps[F] =
+    new MetricsOps[F] {
+      override def increaseActiveRequests(classifier: Option[String]): F[Unit] =
+        metrics.activeRequests
+          .inc(
+            attributes
+              .added(TypedMetricAttributes.classifier(classifier))
+          )
+
+      override def decreaseActiveRequests(classifier: Option[String]): F[Unit] =
+        metrics.activeRequests
+          .dec(
+            attributes
+              .added(TypedMetricAttributes.classifier(classifier))
+          )
+
+      override def recordHeadersTime(
+          method: Method,
+          elapsed: Long,
+          classifier: Option[String],
+      ): F[Unit] =
+        metrics.requestDuration
+          .record(
+            secondsFromNanos(elapsed),
+            attributes
+              .added(TypedMetricAttributes.classifier(classifier))
+              .added(TypedAttributes.httpRequestMethod(method))
+              .added(TypedMetricAttributes.httpPhase(Phase.Headers)),
+          )
+
+      override def recordTotalTime(
+          method: Method,
+          status: Status,
+          elapsed: Long,
+          classifier: Option[String],
+      ): F[Unit] =
+        metrics.requestDuration
+          .record(
+            secondsFromNanos(elapsed),
+            attributes
+              .added(TypedMetricAttributes.classifier(classifier))
+              .added(TypedAttributes.httpRequestMethod(method))
+              .added(TypedAttributes.httpResponseStatusCode(status))
+              .added(TypedMetricAttributes.httpPhase(Phase.Body)),
+          )
+
+      override def recordAbnormalTermination(
+          elapsed: Long,
+          terminationType: TerminationType,
+          classifier: Option[String],
+      ): F[Unit] =
+        metrics.abnormalTerminations
+          .record(
+            secondsFromNanos(elapsed),
+            attributes
+              .added(TypedMetricAttributes.classifier(classifier))
+              .added(TypedMetricAttributes.errorType(terminationType)),
+          )
+
+      private def secondsFromNanos(nanos: Long): Double =
+        nanos / 1000000000.0
+    }
+
+  private def createMetricsCollection[F[_]: Monad: Meter](
+      responseDurationSecondsHistogramBuckets: BucketBoundaries
+  ): F[MetricsCollection[F]] = {
+    val requestDuration: F[Histogram[F, Double]] =
+      Meter[F]
+        .histogram[Double]("http.server.request.duration")
```

Comment:
The `OtelMetrics` can be used with the client and server middleware, so we shouldn't hardcode the `server` classifier here.

What if we provide a category selector:
```scala
sealed trait Category 
object Category {
  case object Server extends Category
  case object Client extends Category
}

def metricsOps[F[_]: Monad: Meter](
    category: Category,
    attributes: Attributes = Attributes.empty,
    responseDurationSecondsHistogramBuckets: BucketBoundaries = DefaultHistogramBuckets,
): F[MetricsOps[F]] =
  for {
    metrics <- createMetricsCollection(category, responseDurationSecondsHistogramBuckets)
  } yield createMetricsOps(metrics, attributes)
  
private def createMetricsCollection[F[_]: Monad: Meter](
    category: Category,
    responseDurationSecondsHistogramBuckets: BucketBoundaries
): F[MetricsCollection[F]] = {
  val prefix = category match {
    case Category.Client => "client"
    case Category.Server => "server"
  }
}
```

Or we can offer different public methods:
```scala
def clientMetricsOps[F[_]: Monad: Meter](...): F[MetricsOps[F]] =

def serverMetricsOps[F[_]: Monad: Meter](...): F[MetricsOps[F]] =
```

Reply:
Good point! I think the two pulic methods look pretty nice because they are self explanatory.